Autopilot/Inc/SensorFusion.hpp

```diff
@@ -120,4 +124,13 @@ Gps_Data_t SF_GetRawGPS();
  */ 
 Altimeter_Data_t SF_GetRawAltimeter();
 
+/**
+    The goal of this function is to return the current track of the plane in degrees, given its velocities in the latitude and longitude direction in m/s.
+
+    * @param[in] latitudeSpeed -> load in the speed of the plane in the N/S direction (change in latitude, m/s)
+    * @param[in] longitudeSpeed -> load in the speed of the plane in the E/W direction (change in longitude, m/s)
+    * @return double - > returns the direction of the resulting velocity/track of the plane
+*/
+double getTrack(double latitudeSpeed, double longitudeSpeed);
```

Comment:
Na, not like this. track is just one of the fields of SFOutput, you'l get it from that structure. The user should never have to call this function. It will just become part of the inside mechanisms of sensor fusion.

To use track, just collect the SFOutput struct from the sensorFusion state and access the track field

Reply:
so I don't even need this function? I thought we agreed sensorFusion would implement this in the future?

Reply:
Yes we did. But this function won't get exposed to the public. It will live inside the guts of sensorFusion with all of Tony's other stuff. When SF_GenerateNewResult is called by Freertos, one of the things that will happen is trackFinding, which will be done by your code. 

But that's Opaque to the user. All they see is a track value that gets set inside the result struct. 

It's a way of abstracting things so users don't have to worry about the complex stuff that goes on in SensorFusion (and it allows us to change SensorFusion guts in the future without having to modify the calling code and such).